examples/sdf2d.py

```diff
@@ -0,0 +1,117 @@
+import taichi as ti
+from math import tau
+ti.init()
+
+N = 512
+img = ti.var(dt=ti.f32, shape=(N, N))
+light_pos = ti.Vector(2, dt=ti.f32, shape=())
+
+@ti.func
+def vres(distance, emission, refl_rate, refl_color):
+    return ti.Vector([distance, emission, refl_rate, refl_color])
+
+@ti.func
+def vec2(x, y):
+    return ti.Vector([x, y])
+
+@ti.func
+def vec3(x, y, z):
+    return ti.Vector([x, y, z])
+
+@ti.func
+def union(a, b):
+    if a[0] > b[0]:
+        a = b
+    return a
+
+@ti.func
+def intersect(a, b):
+    if a[0] < b[0]:
+        a = b
+    return a
+
+@ti.func
+def reflect(r, n):
```

Comment:
FYI I just added this https://github.com/taichi-dev/taichi/blob/5db9f22074f8da749edf8869b8854390fe117f75/examples/renderer_utils.py#L23-L25

Reply:
Thanks for the useful stuff! Btw #838 includes some of these too.